### PR TITLE
Phase 6c-prep follow-up: mount park / unpark / get_park_state / abort_slew

### PR DESCRIPTION
**Pull request by ivonnyssen**
## Summary

Adds the four mount-control MCP tools deferred from Phase 6c-prep (PR #133), completing the singular-mount tool surface for `center_on_target` (6c-3) and safety-shutdown workflows.

**Tools (all four take no parameters):**
- **`park`** — invoke ASCOM `Park`, poll `Slewing` until idle (300 s deadline), verify `AtPark == true`. Per ASCOM, a successful park clears `Tracking`. Unlike `slew`, `park` does NOT auto-abort on timeout — a partially-completed park is closer to safe than an aborted one (the mount is actively trying to reach a known safe position; aborting leaves it in an unknown state mid-traversal). Callers wanting to interrupt a stuck park use `abort_slew`.
- **`unpark`** — returns immediately (most drivers just clear `AtPark` — no motion). Does NOT auto-enable `Tracking`; caller calls `set_tracking` explicitly before slewing.
- **`get_park_state`** — `{ at_park, can_park, can_unpark }`. Fails loud on `AtPark` read errors (load-bearing field).
- **`abort_slew`** — aborts an in-progress slew or park. Per ASCOM, only valid while `Slewing == true`; the natural Alpaca error propagates otherwise.

**Implementation notes:**
- Extracted the `Slewing` poll loop from `do_slew_blocking` into a shared `poll_slewing_until_idle` helper (returns a typed `PollIdleError` so `do_slew_blocking` best-effort-aborts on `Timeout` while `do_park_blocking` surfaces the timeout untouched). Eliminates ~10 lines of duplication.
- `MockTelescope` grew `at_park` / `can_park` / `can_unpark` fields plus six `fail_*` injection points. Defaults remain happy-path (capable, not parked).

## Test plan

- [x] `cargo rail run --profile commit -q` — 254/254 affected-package tests pass.
- [x] `cargo test --all-features --test bdd -p rp` — 192/192 BDD scenarios green (was 187, +5 new park scenarios).
- [x] **+14 new mcp.rs unit tests**: 5 park (incl. `start_paused` timeout test that verifies park does NOT auto-abort), 3 unpark, 2 get_park_state, 3 abort_slew, plus the completes-but-not-at-park edge case.
- [x] **+5 new BDD scenarios** on `mount.feature`: park happy path verified via `get_park_state`, park no-mount-configured, unpark round-trip (park → unpark → at_park: false), get_park_state field-presence, abort_slew no-mount-configured. Catalog scenario extended with all four new tools.
- [x] `cargo fmt` clean.
- [x] No new workspace deps; no `bazel mod tidy` needed.

## Files

- `services/rp/src/mcp.rs` — `do_park_blocking` + extracted `poll_slewing_until_idle` helper + 4 tools + `MockTelescope` extensions + 14 unit tests.
- `services/rp/tests/features/mount.feature` — catalog extended; 5 new scenarios.
- `services/rp/tests/bdd/steps/mount_steps.rs` — When/Then steps for the 4 new tools.
- `docs/services/rp.md` — Built-in Tools — Hardware table gains 4 rows.
- `docs/plans/image-evaluation-tools.md` — Phase 6c-prep deferred-list updated (removed `park`/`unpark` and `abort_slew`; added a follow-up-landed note; `set_park` stays deferred).

## Out of scope (still deferred)

- `set_park` (sets the current position as the new park position) — power-user feature with no workflow consumer in sight.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: services/rp/src/mcp.rs [hunk 1]
```diff
@@ -1053,6 +1066,73 @@ impl McpHandler {
...
+                return Err("timeout waiting for mount to park".to_string());
+            }
+            Err(PollIdleError::Read(e)) => {
+                return Err(format!("error polling mount slewing: {}", e));
+            }
+        }
+
+        match mount.at_park().await {
+            Ok(true) => Ok(()),
+            Ok(false) => Err("mount completed slew but is not at park".to_string()),
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp/src/mcp.rs]: Agreed — fixed in 086b62e. The error string is now `"park completed but mount is not at park"`, naming the failing operation explicitly so MCP callers don't have to mentally translate "slew" → park. Updated the matching unit-test substring assertion as well.

File: services/rp/tests/features/mount.feature [hunk 2]
```diff
@@ -149,3 +153,47 @@ Feature: Mount tools
...
     Then the tool call should succeed
     And the get_tracking result tracking should be false
+
+  Scenario: park stows the mount and reports at_park == true
+    Given a running Alpaca simulator
+    And rp is running with a mount on the simulator
+    And an MCP client connected to rp
+    When the MCP client calls "park"
+    And the MCP client calls "get_park_state"
+    Then the tool call should succeed
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp/tests/features/mount.feature]: Agreed — fixed in 086b62e. The scenario now inserts `Then the tool call should succeed` between the `park` and `get_park_state` When steps. A regression where `park` returns an MCP error after parking the simulator now fails this scenario directly, instead of being masked by the trailing `get_park_state` overwrite of `last_tool_result`.

File: services/rp/tests/features/mount.feature [hunk 3]
```diff
@@ -149,3 +153,47 @@ Feature: Mount tools
...
+    And the error message should contain "no mount configured"
+
+  Scenario: unpark clears the mount's parked flag
+    Given a running Alpaca simulator
+    And rp is running with a mount on the simulator
+    And an MCP client connected to rp
+    When the MCP client calls "park"
+    And the MCP client calls "unpark"
+    And the MCP client calls "get_park_state"
+    Then the tool call should succeed
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp/tests/features/mount.feature]: Agreed — fixed in 086b62e. The unpark scenario now asserts success after each chained call (park, get_park_state, unpark, get_park_state) and observes `at_park == true` after park before continuing — so a no-op park or no-op unpark fails the scenario at the relevant step rather than silently passing on the final state check.

File: services/rp/tests/features/mount.feature [hunk 4]
```diff
@@ -149,3 +153,47 @@ Feature: Mount tools
...
+    And the get_park_state result at_park should be false
+
+  Scenario: get_park_state returns at_park, can_park, and can_unpark
+    Given a running Alpaca simulator
+    And rp is running with a mount on the simulator
+    And an MCP client connected to rp
+    When the MCP client calls "get_park_state"
+    Then the tool call should succeed
```
[the remark this answers is not in the document]
Comment (ivonnyssen) [reply services/rp/tests/features/mount.feature:201]: Agreed — fixed in 086b62e. The scenario now asserts the `at_park` field as well (matching the title's promise of all three fields). Added an `unpark` setup step at the start so `at_park == false` is deterministic regardless of whether earlier scenarios in the feature parked the singleton OmniSim mount — scenario isolation rather than relying on prior-scenario side-effects.
